packages/squiggle-lang/__tests__/Distributions/AlgebraicCombination_test.res

```diff
@@ -0,0 +1,313 @@
+open Jest
+open Expect
+open TestHelpers
+
+let {
+  normalDist5,  // mean=5, stdev=2
+  normalDist10,  // mean=10, stdev=2
+  normalDist20,  // mean=20, stdev=2
+  normalDist,  // mean=5; stdev=2
+  uniformDist,  // low=9; high=10
+  betaDist,  // alpha=2; beta=5
+  lognormalDist,  // mu=0; sigma=1
+  cauchyDist,  // local=1; scale=1
+  triangularDist,  // low=1; medium=2; high=3;
+  exponentialDist,  // rate=2
+} = module(GenericDist_Fixtures)
+
+let {
+    algebraicAdd, 
+    algebraicMultiply, 
+    algebraicDivide, 
+    algebraicSubtract, 
+    algebraicLogarithm, 
+    algebraicPower
+} = module(DistributionOperation.Constructors)
+
+let algebraicAdd = algebraicAdd(~env)
+let algebraicMultiply = algebraicMultiply(~env)
+let algebraicDivide = algebraicDivide(~env)
+let algebraicSubtract = algebraicSubtract(~env)
+let algebraicLogarithm = algebraicLogarithm(~env)
+let algebraicPower = algebraicPower(~env)
+
+describe("(Algebraic) addition of distributions", () => {
+
+    describe("mean", () => {
+        test("normal(mean=5) + normal(mean=20)", () => {
+            normalDist5
+            -> algebraicAdd(normalDist20)
+            -> E.R2.fmap(GenericDist_Types.Constructors.UsingDists.mean)
+            -> E.R2.fmap(run)
+            -> E.R2.fmap(toFloat)
+            -> E.R.toExn
+            -> expect
+            -> toBe(Some(2.5e1))
+        })
+
+        test("uniform(low=9, high=10) + beta(alpha=2, beta=5)", () => {
+            // let uniformMean = (9.0 +. 10.0) /. 2.0
+            // let betaMean = 1.0 /. (1.0 +. 5.0 /. 2.0)
+            let received = uniformDist 
+                -> algebraicAdd(betaDist)
+                -> E.R2.fmap(GenericDist_Types.Constructors.UsingDists.mean)
+                -> E.R2.fmap(run)
+                -> E.R2.fmap(toFloat)
+                -> E.R.toExn
+            switch received {
+                | None => "algebraicAdd has" -> expect -> toBe("failed")
+                // This is nondeterministic, we could be in a situation where ci fails but you click rerun and it passes, which is bad. 
```

Comment:
Yea, this isn't good. If it fails more than 0.01% of the time, we really should figure out something else here.

Reply:
frankly with precision down around 0-2 digits I think the failure rate is fairly negligible. it's when I try to push to 3 or 4 it gets to be too much.